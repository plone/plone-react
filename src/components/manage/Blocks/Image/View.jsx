/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import Image from '@plone/volto/components/theme/Image/Image';

import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <Image
              image={data.url}
              alt={data.alt}
              className={cx({
                'full-width': data.align === 'full',
                large: data.size === 'l',
                medium: data.size === 'm',
                small: data.size === 's',
              })}
            />
          );
          if (data.href) {
            if (!isInternalURL(data.href)) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.href}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={flattenToAppURL(data.href)}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
