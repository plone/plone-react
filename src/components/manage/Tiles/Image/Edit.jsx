/**
 * Edit image tile.
 * @module components/manage/Tiles/Image/Edit
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { readAsDataURL } from 'promise-file-reader';
import { Button, Dimmer, Input, Loader, Message } from 'semantic-ui-react';
import { defineMessages, injectIntl, intlShape } from 'react-intl';
import cx from 'classnames';
import Dropzone from 'react-dropzone';

import { settings } from '~/config';
import withObjectBrowser from '../../Sidebar/ObjectBrowser';

import { Icon, ImageSidebar, SidebarPortal } from '@plone/volto/components';
import { createContent } from '@plone/volto/actions';
import { flattenToAppURL, getBaseUrl } from '@plone/volto/helpers';

import imageTileSVG from './tile-image.svg';
import clearSVG from '@plone/volto/icons/clear.svg';
import navTreeSVG from '@plone/volto/icons/nav.svg';
import aheadSVG from '@plone/volto/icons/ahead.svg';

const messages = defineMessages({
  ImageTileInputPlaceholder: {
    id: 'Browse the site, drop an image, or type an URL',
    defaultMessage: 'Browse the site, drop an image, or type an URL',
  },
});

/**
 * Edit image tile class.
 * @class Edit
 * @extends Component
 */
class Edit extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    selected: PropTypes.bool.isRequired,
    tile: PropTypes.string.isRequired,
    index: PropTypes.number.isRequired,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    content: PropTypes.objectOf(PropTypes.any).isRequired,
    request: PropTypes.shape({
      loading: PropTypes.bool,
      loaded: PropTypes.bool,
    }).isRequired,
    pathname: PropTypes.string.isRequired,
    onChangeTile: PropTypes.func.isRequired,
    onSelectTile: PropTypes.func.isRequired,
    onDeleteTile: PropTypes.func.isRequired,
    onFocusPreviousTile: PropTypes.func.isRequired,
    onFocusNextTile: PropTypes.func.isRequired,
    handleKeyDown: PropTypes.func.isRequired,
    createContent: PropTypes.func.isRequired,
    openObjectBrowser: PropTypes.func.isRequired,
    intl: intlShape.isRequired,
  };

  state = {
    uploading: false,
    url: '',
  };

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    if (this.props.selected) {
      this.node.current.focus();
    }
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (
      this.props.request.loading &&
      nextProps.request.loaded &&
      this.state.uploading
    ) {
      this.setState({
        uploading: false,
      });
      this.props.onChangeTile(this.props.tile, {
        ...this.props.data,
        url: nextProps.content['@id'],
      });
    }

    if (nextProps.selected) {
      this.node.current.focus();
    }
  }

  /**
   * Upload image handler (not used), but useful in case that we want a button
   * not powered by react-dropzone
   * @method onUploadImage
   * @returns {undefined}
   */
  onUploadImage = ({ target }) => {
    const file = target.files[0];
    this.setState({
      uploading: true,
    });
    readAsDataURL(file).then(data => {
      const fields = data.match(/^data:(.*);(.*),(.*)$/);
      this.props.createContent(getBaseUrl(this.props.pathname), {
        '@type': 'Image',
        title: file.name,
        image: {
          data: fields[3],
          encoding: fields[2],
          'content-type': fields[1],
          filename: file.name,
        },
      });
    });
  };

  /**
   * Align tile handler
   * @method onAlignTile
   * @param {string} align Alignment option
   * @returns {undefined}
   */
  onAlignTile(align) {
    this.props.onChangeTile(this.props.tile, {
      ...this.props.data,
      align,
    });
  }

  /**
   * Change url handler
   * @method onChangeUrl
   * @param {Object} target Target object
   * @returns {undefined}
   */
  onChangeUrl = ({ target }) => {
    this.setState({
      url: target.value,
    });
  };

  /**
   * Submit url handler
   * @method onSubmitUrl
   * @param {object} e Event
   * @returns {undefined}
   */
  onSubmitUrl = () => {
    this.props.onChangeTile(this.props.tile, {
      ...this.props.data,
      url: this.state.url,
    });
  };

  /**
   * Drop handler
   * @method onDrop
   * @param {array} files File objects
   * @returns {undefined}
   */
  onDrop = file => {
    this.setState({
      uploading: true,
    });

    readAsDataURL(file[0]).then(data => {
      const fields = data.match(/^data:(.*);(.*),(.*)$/);
      this.props.createContent(getBaseUrl(this.props.pathname), {
        '@type': 'Image',
        title: file[0].name,
        image: {
          data: fields[3],
          encoding: fields[2],
          'content-type': fields[1],
          filename: file[0].name,
        },
      });
    });
  };

  /**
   * Keydown handler on Variant Menu Form
   * This is required since the ENTER key is already mapped to a onKeyDown
   * event and needs to be overriden with a child onKeyDown.
   * @method onKeyDownVariantMenuForm
   * @param {Object} e Event object
   * @returns {undefined}
   */
  onKeyDownVariantMenuForm = e => {
    if (e.key === 'Enter') {
      e.preventDefault();
      e.stopPropagation();
      this.onSubmitUrl();
    } else if (e.key === 'Escape') {
      e.preventDefault();
      e.stopPropagation();
      // TODO: Do something on ESC key
    }
  };

  node = React.createRef();

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div
        role="presentation"
        onClick={() => this.props.onSelectTile(this.props.tile)}
        className={cx(
          'tile image align',
          {
            selected: this.props.selected,
            center: !Boolean(this.props.data.align),
          },
          this.props.data.align,
        )}
        onKeyDown={e =>
          this.props.handleKeyDown(
            e,
            this.props.index,
            this.props.tile,
            this.node.current,
          )
        }
        ref={this.node}
      >
        {this.props.selected && !!this.props.data.url && (
          <div className="toolbar">
            {this.props.appendActions && <>{this.props.appendActions}</>}
            {this.props.detached && this.props.appendActions && (
              <div className="separator" />
            )}
            <Button.Group>
              <Button
                icon
                basic
                onClick={() =>
                  this.props.onChangeTile(this.props.tile, {
                    ...this.props.data,
                    url: '',
                  })
                }
              >
                <Icon name={clearSVG} size="24px" color="#e40166" />
              </Button>
            </Button.Group>
            {this.props.appendSecondaryActions && (
              <>{this.props.appendSecondaryActions}</>
            )}
          </div>
        )}
        {this.props.selected &&
          !this.props.data.url &&
          this.props.appendSecondaryActions && (
            <div className="toolbar">{this.props.appendSecondaryActions}</div>
          )}
        {this.props.data.url ? (
          <p>
            <img
              src={
                this.props.data.url.includes(settings.apiPath)
                  ? `${flattenToAppURL(this.props.data.url)}/@@images/image`
                  : this.props.data.url
              }
              alt=""
            />
          </p>
        ) : (
          <div>
            <Dropzone onDrop={this.onDrop} className="dropzone">
              <Message>
                {this.state.uploading && (
                  <Dimmer active>
                    <Loader indeterminate>Uploading image</Loader>
                  </Dimmer>
                )}
                <center>
                  <img src={imageTileSVG} alt="" />
                  <div className="toolbar-inner">
                    <Button.Group>
                      <Button
                        basic
                        icon
                        onClick={e => {
                          e.stopPropagation();
                          this.props.openObjectBrowser();
                        }}
                      >
                        <Icon name={navTreeSVG} size="24px" />
                      </Button>
                    </Button.Group>
                    <Input
                      onKeyDown={this.onKeyDownVariantMenuForm}
                      onChange={this.onChangeUrl}
                      placeholder={this.props.intl.formatMessage(
                        messages.ImageTileInputPlaceholder,
                      )}
                      // Prevents propagation to the Dropzone and the opening
                      // of the upload browser dialog
                      onClick={e => e.stopPropagation()}
                    />
                    {this.state.url && (
                      <Button.Group>
                        <Button basic className="cancel">
                          <Icon name={clearSVG} size="30px" />
                        </Button>
                      </Button.Group>
                    )}
                    <Button.Group>
                      <Button basic primary>
                        <Icon name={aheadSVG} size="30px" />
                      </Button>
                    </Button.Group>
                  </div>
                </center>
              </Message>
            </Dropzone>
          </div>
        )}
        <SidebarPortal selected={this.props.selected}>
          <ImageSidebar {...this.props} />
        </SidebarPortal>
      </div>
    );
  }
}

export default compose(
  withObjectBrowser,
  injectIntl,
  connect(
    state => ({
      request: state.content.create,
      content: state.content.data,
    }),
    { createContent },
  ),
)(Edit);
