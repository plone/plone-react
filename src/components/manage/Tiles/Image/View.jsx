/**
 * View image tile.
 * @module components/manage/Tiles/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { settings } from '~/config';

import { flattenToAppURL } from '@plone/volto/helpers';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'tile image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <img
              src={
                data.url.includes(settings.apiPath)
                  ? `${flattenToAppURL(data.url)}/@@images/image`
                  : data.url
              }
              alt={data.alt || ''}
            />
          );
          if (data.external) {
            const isReallyExternal =
              (data.external.startsWith('http') ||
                data.external.startsWith('https')) &&
              !data.external.includes(settings.apiPath);

            if (isReallyExternal) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.external}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={data.external.replace(settings.apiPath, '')}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else if (data.href) {
            return (
              <Link
                to={data.href}
                target={data.openLinkInNewTab ? '_blank' : null}
              >
                {image}
              </Link>
            );
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
