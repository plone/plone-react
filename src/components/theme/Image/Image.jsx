import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getImageAttributes } from '@plone/volto/helpers';

/**
 * Image component
 * @param {object | string} image - Plone image as object or url
 * @param {string} alt - Alternative text for image
 * @param {string} className - CSS class attribute
 * @param {string} role - img role attribute
 * @param {boolean} critical - whether to lazy load the image
 * @param {number} maxSize - maximum size to render
 * @param {boolean} useOriginal - whether to render original size
 */
const Image = ({
  image,
  alt = '',
  className,
  role = 'img',
  critical = false,
  maxSize,
  useOriginal = false,
  ...imageProps
}) => {
  const { src, srcSet } = getImageAttributes(image, { maxSize, useOriginal });
  const [isClient, setIsClient] = useState(false);
  const [srcset, setSrcset] = useState(
    critical && srcSet ? srcSet.join(', ') : null,
  );

  useEffect(() => {
    setIsClient(true);
    if (!srcset && srcSet?.length > 0) setSrcset(srcSet.join(', '));
  }, [srcSet, srcSet?.length, srcset]);

  return (
    <>
      <picture>
        {srcset && (isClient || critical) && <source srcSet={srcset} />}
        <img
          src={src}
          alt={alt}
          className={className}
          role={role}
          loading="lazy"
          style={{ width: '100%', objectFit: 'cover' }}
          {...imageProps}
        />
      </picture>
      {!critical && (
        <noscript
          dangerouslySetInnerHTML={{
            __html: `
              <img
                src="${src}"
                ${srcSet?.length && `srcset="${srcSet.join(', ')}"`}
                alt="${alt}"
                className="${className}"
                role="${role}"
                loading="lazy"
                style="width: 100%; object-fit: cover;"
            `,
          }}
        />
      )}
    </>
  );
};

Image.propTypes = {
  image: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  alt: PropTypes.string,
  className: PropTypes.string,
  role: PropTypes.string,
  critical: PropTypes.bool,
  maxSize: PropTypes.number,
  useOriginal: PropTypes.bool,
};

export default Image;
