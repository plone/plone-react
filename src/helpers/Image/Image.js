import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';
import config from '@plone/volto/registry';

/**
 * Get src-set list from image
 * @param {object | string} image - Image content object or url
 * @param {object} options - { maxSize [number]: maximum size to render, useOriginal [boolean]: whether to render original img }
 */
export const getImageAttributes = (
  image,
  { maxSize = 10000, useOriginal = false } = {},
) => {
  const imageScales = config.settings.imageScales;
  const minSize = Object.keys(imageScales).reduce((minSize, scale) => {
    if (!minSize || imageScales[scale] < imageScales[minSize]) {
      return scale;
    }

    return minSize;
  }, null);

  let attrs = {};
  let imageType =
    Object.prototype.toString.call(image) === '[object Object]' &&
    image.scales &&
    Object.keys(image.scales).length > 0
      ? 'imageObject'
      : typeof image === 'string' && isInternalURL(image)
      ? 'internalUrl'
      : 'external';

  switch (imageType) {
    // Scales object from Plone restapi
    // ideal use of Plone images
    case 'imageObject':
      let sortedScales = Object.values(image.scales)
        .filter((scale) => scale.width <= maxSize)
        .filter((scale) => scale.width < image.width) // avoid duplicates if image is small and original is smaller than scale
        .sort((a, b) => {
          if (a.width > b.width) return 1;
          else if (a.width < b.width) return -1;
          else return 0;
        });
      attrs.src = sortedScales[0].download;
      attrs.srcSet = sortedScales.map(
        (scale) => `${flattenToAppURL(scale.download)} ${scale.width}w`,
      );

      if (useOriginal)
        attrs.srcSet = attrs.srcSet.concat(
          `${flattenToAppURL(image.download)} ${image.width}w`,
        );
      break;

    // Internal URL
    case 'internalUrl':
      let baseUrl = `${flattenToAppURL(image.split('/@@images')[0])}${
        image.endsWith('/') ? '' : '/'
      }@@images/image`;
      attrs.src = `${baseUrl}/${minSize}`;
      attrs.srcSet = Object.keys(imageScales).reduce((srcSet, scale) => {
        return [...srcSet, `${baseUrl}/${scale} ${imageScales[scale]}w`];
      }, []);

      if (useOriginal) attrs.srcSet = attrs.srcSet.concat(`${baseUrl} 1200w`); // expect that is for desktop screens, I don't have actual size
      break;

    // External URL or other element
    default:
      attrs.src = typeof image === 'string' ? image : null;
      break;
  }

  return attrs;
};
