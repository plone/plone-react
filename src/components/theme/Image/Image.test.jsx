import React from 'react';
import renderer from 'react-test-renderer';

import Image from './Image';

const ploneImage = {
  download: 'http://localhost:8080/Plone/test-images/@@images/image',
  width: 1920,
  scales: {
    icon: {
      download: 'http://localhost:8080/Plone/test-images/@@images/image/icon',
      width: 32,
    },
    large: {
      download: 'http://localhost:8080/Plone/test-images/@@images/image/large',
      width: 768,
    },
    listing: {
      download:
        'http://localhost:8080/Plone/test-images/@@images/image/listing',
      width: 16,
    },
    mini: {
      download: 'http://localhost:8080/Plone/test-images/@@images/image/mini',
      width: 200,
    },
    preview: {
      download:
        'http://localhost:8080/Plone/test-images/@@images/image/preview',
      width: 400,
    },
    thumb: {
      download: 'http://localhost:8080/Plone/test-images/@@images/image/thumb',
      width: 128,
    },
    tile: {
      download: 'http://localhost:8080/Plone/test-images/@@images/image/tile',
      width: 64,
    },
  },
};

describe('Image', () => {
  it('renders image with default values and image url', () => {
    const component = renderer.create(
      <Image image="http://localhost:8080/Plone/photo.png" />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders image with props and image url', () => {
    const component = renderer.create(
      <Image
        image="http://localhost:8080/Plone/photo.png"
        alt="Photo"
        className="photo-image"
        role="presentation"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders image with plone image object', () => {
    const component = renderer.create(
      <Image
        image={ploneImage}
        alt="Photo"
        className="photo-image"
        role="presentation"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders image with external url', () => {
    const component = renderer.create(
      <Image
        image="https://picsum.photos/200/300"
        alt="External picsum photo"
      />,
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
