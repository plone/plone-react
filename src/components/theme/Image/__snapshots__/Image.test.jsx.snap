// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders image with default values and image url 1`] = `
Array [
  <picture>
    <img
      alt=""
      loading="lazy"
      role="img"
      src="/photo.png/@@images/image/listing"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"/photo.png/@@images/image/listing\\"
                srcset=\\"/photo.png/@@images/image/large 768w, /photo.png/@@images/image/preview 400w, /photo.png/@@images/image/mini 200w, /photo.png/@@images/image/thumb 128w, /photo.png/@@images/image/tile 64w, /photo.png/@@images/image/icon 32w, /photo.png/@@images/image/listing 16w\\"
                alt=\\"\\"
                className=\\"undefined\\"
                role=\\"img\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;

exports[`Image renders image with external url 1`] = `
Array [
  <picture>
    <img
      alt="External picsum photo"
      loading="lazy"
      role="img"
      src="https://picsum.photos/200/300"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"https://picsum.photos/200/300\\"
                undefined
                alt=\\"External picsum photo\\"
                className=\\"undefined\\"
                role=\\"img\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;

exports[`Image renders image with plone image object 1`] = `
Array [
  <picture>
    <img
      alt="Photo"
      className="photo-image"
      loading="lazy"
      role="presentation"
      src="http://localhost:8080/Plone/test-images/@@images/image/listing"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"http://localhost:8080/Plone/test-images/@@images/image/listing\\"
                srcset=\\"/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w, /test-images/@@images/image/preview 400w, /test-images/@@images/image/large 768w\\"
                alt=\\"Photo\\"
                className=\\"photo-image\\"
                role=\\"presentation\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;

exports[`Image renders image with plone image object and opt props 1`] = `
Array [
  <picture>
    <img
      alt="Photo"
      className="photo-image"
      loading="lazy"
      role="presentation"
      src="http://localhost:8080/Plone/test-images/@@images/image/listing"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"http://localhost:8080/Plone/test-images/@@images/image/listing\\"
                srcset=\\"/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w\\"
                alt=\\"Photo\\"
                className=\\"photo-image\\"
                role=\\"presentation\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;

exports[`Image renders image with plone image object and original 1`] = `
Array [
  <picture>
    <source
      srcSet="/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w, /test-images/@@images/image/preview 400w, /test-images/@@images/image/large 768w, /test-images/@@images/image 1920w"
    />
    <img
      alt="Photo"
      className="photo-image"
      loading="lazy"
      role="presentation"
      src="http://localhost:8080/Plone/test-images/@@images/image/listing"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"http://localhost:8080/Plone/test-images/@@images/image/listing\\"
                srcset=\\"/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w, /test-images/@@images/image/preview 400w, /test-images/@@images/image/large 768w, /test-images/@@images/image 1920w\\"
                alt=\\"Photo\\"
                className=\\"photo-image\\"
                role=\\"presentation\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;

exports[`Image renders image with props and image url 1`] = `
Array [
  <picture>
    <source
      srcSet="/photo.png/@@images/image/large 768w, /photo.png/@@images/image/preview 400w, /photo.png/@@images/image/mini 200w, /photo.png/@@images/image/thumb 128w, /photo.png/@@images/image/tile 64w, /photo.png/@@images/image/icon 32w, /photo.png/@@images/image/listing 16w"
    />
    <img
      alt="Photo"
      className="photo-image"
      loading="lazy"
      role="presentation"
      src="/photo.png/@@images/image/listing"
      style={
        Object {
          "objectFit": "cover",
          "width": "100%",
        }
      }
    />
  </picture>,
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "
              <img
                src=\\"/photo.png/@@images/image/listing\\"
                srcset=\\"/photo.png/@@images/image/large 768w, /photo.png/@@images/image/preview 400w, /photo.png/@@images/image/mini 200w, /photo.png/@@images/image/thumb 128w, /photo.png/@@images/image/tile 64w, /photo.png/@@images/image/icon 32w, /photo.png/@@images/image/listing 16w\\"
                alt=\\"Photo\\"
                className=\\"photo-image\\"
                role=\\"presentation\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
      }
    }
  />,
]
`;
